plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/versions/VersionsPropertiesModel.kt

```diff
@@ -0,0 +1,79 @@
+package de.fayard.refreshVersions.core.internal.versions
+
+internal data class VersionsPropertiesModel(
+    val preHeaderContent: String,
+    val generatedByVersion: String,
+    val sections: List<Section>
+) {
+    init {
+        if (preHeaderContent.isNotEmpty()) require(preHeaderContent.endsWith('\n'))
+        preHeaderContent.lineSequence().forEach { if (it.isNotBlank()) it.mustBeACommentLine() }
+    }
+
+    sealed class Section {
+
+        data class Comment(val lines: String) : Section() {
+            init {
+                lines.lineSequence().forEach { it.mustBeACommentLine() }
+            }
+        }
+
+        data class VersionEntry(
+            val leadingCommentLines: List<String>,
+            val key: String,
+            val currentVersion: String,
+            val availableUpdates: List<String>,
+            val trailingCommentLines: List<String>
+        ) : Section() {
```

Comment:
For unit tests, my philosophy is that the smaller there are, the better. 
Instead of having `private fun VersionsPropertiesModel.Section.toText(stringBuilder)`,
I would simply override the `toString()` method for every subclass of section.
This way a test can be written simply as 

```kotlin
val entry = VersionEntry(key = "version.kotlin", currentVersion = "1.4")
"$entry" shouldBe "version.kotlin=1.4"
```

Reply:
`toString()` is more targeted at debug printing when it's not about `CharSequence` or numbers. I'm afraid it would be confusing to have the `toString()` be part of the "business" logic, especially in a `data class`.

I don't disagree on making tests simple, but I don't think smaller is always simpler.

I'll see if the tests can be made simpler though.

Reply:
I'm fine with `toString()` but if not just declare an abstract method `toText() : CharSequence` on the `Section` class and the same thing works

Reply:
Regarding the current `toText()` function, why do you think it'd be better to have it as an abstract function in the `Section` sealed class instead of an extension as it currently is?

I personally chose the extension function approach to not overcrowd the model file and limit it to arguments checking and related constants, putting the read and write logic externally.

Reply:
It's fine to have one single `Section.toText()` method.